generators/app/templates/client/app/components/model-input-group/model-input/input.html

```diff
@@ -1,8 +1,21 @@
 <md-input-container flex>
-  <label>{{fieldDef.desc}}</label>
+  <label>
+    {{fieldDef.desc}}<span ng-if="fieldDef.validators.required">*</span>
+  </label>
+
   <input name="{{fieldDef.name}}"
          ng-model="ngModel[fieldDef.name]"
          ng-change="ngChange(fieldDef.name, ngModel[fieldDef.name])"
          type="{{fieldDef.type}}"
-         ng-required="fieldDef.required" />
+         mongoose-error="{{fieldDef.validators.mongoose.value}}"
```

Comment:
The mongoose-error directive extends a form or ng-form with a `setResponseErrors` method, you may find in the [catch cases of client controllers](https://github.com/michaelkrone/generator-material-app/blob/develop/generators/app/templates/client/app/admin(auth)/user/create/create.controller.js#L111), which will take an error response: 
 ```js
if (form && err) {
 form.setResponseErrors(err);
}
```
`setResponseErrors` parses the request for a mongoose error message and applies the mongoose validation results to the field if the field's name matches the database document property. So mongoose validation errors can be handled in the ng-messages without any further doing.

Since we cannot revalidate the complete [mongoose validation on the client](http://mongoosejs.com/docs/browser.html), the directive sets the mongoose error state to false on any user input.